packages/fuels/src/cli/commands/deploy/createWallet.ts

```diff
@@ -11,7 +11,19 @@ export async function createWallet(providerUrl: string, privateKey?: string) {
     throw new Error('You must provide a privateKey via config.privateKey or env PRIVATE_KEY');
   }
 
-  const provider = await Provider.create(providerUrl);
+  try {
+    const provider = await Provider.create(providerUrl);
 
-  return Wallet.fromPrivateKey(pvtKey, provider);
+    return Wallet.fromPrivateKey(pvtKey, provider);
+  } catch (e) {
+    const error = e as Error & { cause?: { code: string } };
+
+    if (error.cause?.code === 'ECONNREFUSED') {
+      throw new Error(
```

Comment:
Should we replace this with a `FuelError` to keep the standards?

Reply:
I'm not really sure. I see `FuelError` as a way of conveying domain-related errors and this here is more a technical issue of the initial `TypeError: fetch failed` not being descriptive enough and us just decorating it a bit for quality of life. Not every error deserves to be a `FuelError` :smiley:. What do you think?

Reply:
I see what you mean, and I agree, but leaving the decision to add or not to add open to interpretation could lead to unintentional mistakes in the future. Besides, the error message we're using is somewhat domain-specific. 

I'd vote to use `FuelError` for every error _we_ throw. The extra decoration you mentioned requires _us_ to throw the error, so I'd add it to the list of error codes, especially after the whole discussion we had about it today.

On the same line as what you said, having an error code pre-defined would make it easier for consumers to try/catch things predictably, improving the _quality of life_. :)

Reply:
Anything for the user to be able to quickly identify, and debug the errors they might face would be benefical. I'd also vote for `FuelError` every error within our codebase :)

Do we have a descicion log for capturing this information? (once agreed upon obviously)

Reply:
https://github.com/FuelLabs/fuels-ts/pull/1854/commits/b786b9bec7d0eab6009e51f5e25db3b766600d53